### DISCUSSION TITLE
dsl matcher not work correctly with payloads

### DISCUSSION THREAD

**Discussion by 0xRyuzak1** [projectdiscovery/nuclei#4125] [Q&A]
I Created a new Template to do the following :
- Iterate through Specific HTTP Verbs like ( GET , TRACE , etc ) while also iterate through some paths
- For each path and verb make two requests and check if the following happens :
  - First request return status not equal to 404 
  - Sec request return status equal to 404 

Its working correct for the first path and verb but then i found that the status_code_1 and status_code_2 keep holding the result from the previous two requests which break the logic so can you please help me 

```
id: test

info:
  name: test Vulnerability
  author: anyone
  severity: Medium

http:
  - raw:
      - |
        {{Verb}} {{BaseURL}}/{{MagicFinalPart}} HTTP/1.1
        Host: {{Hostname}}     
      - |
        {{Verb}} {{BaseURL}}/{{MagicFinalPart}}?payload=1 HTTP/1.1
        Host: {{Hostname}}
        
    payloads:
      Verb:
        - "GET"
        - "POST"

      MagicFinalPart:
        - 'Path1'
        - 'Path2'
        - 'Path3'


    attack: clusterbomb
    stop-at-first-match: true
    matchers:
      - type: dsl
        dsl:
          - "status_code_1 != 404 && status_code_2 == 404"
        
```

**Comment by MetzinAround** [comment]
Hi @0xRyuzak1 !! I'm gonna ask @olearycrew 

Brendan, I was looking at this and can't figure out how to get the variables to reset between requests. Any idea?

**Reply by olearycrew** [reply]
Interesting. I'm worried that each new request is getting a `status_code` variable - as in the third request it makes is `status_code_3`

@Mzack9999 or @ehsandeep could confirm this.

**Reply by 0xRyuzak1** [reply]
Hi @MetzinAround @olearycrew Actually this not what happened in this case there are only a two status_code are used `status_code_1` and `status_code_2` which is not the problem the problem is that the status_code variable is holding the previous results after the condition checks for the first time which will break the logic and i can't solve this :(

**Reply by MetzinAround** [reply]
~I'm not seeing where `status_code_1` and `status_code_2` are declared except down in the matchers.~

~Where do they get initially declared?~

Striking out unnecessary question. We're going to have to check with our folks tagged above to get help on this one. Sorry!

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
